### ISSUE TITLE
forward-auth plugin request timeout

### ISSUE THREAD

**Issue by dashixiongsk** [apache/apisix#11050]
### Current Behavior

The forward auth plugin uses post requests, request_headers configures Content Type request headers, and requests timeout

### Steps to Reproduce

this is my config:

```
 "forward-auth": {
      "_meta": {
        "disable": false
      },
      "client_headers": [
        "Location"
      ],
      "request_headers": [
        "Authorization", "Content-Type", Content-Length
      ],
      "request_method": "POST",
      "ssl_verify": false,
      "upstream_headers": [
        "x-offcn-gateway-id",
      ],
      "uri": "http://127.0.0.1:9080/checkAuth"
    }
```

I timed out using the Postman request interface。 When I removed the Content Type from request_headers, it did not time out but reported an error: {"code": 400, "reason": "CODEC", "message": "unregister Content Type:", "metadata": {}}

### Environment

- APISIX version (run `apisix version`):
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

**Comment by theweakgod** [comment]
> "Authorization", "Content-Type", Content-Length
> "x-offcn-gateway-id",

There is something wrong with this json format. Please make sure the json format is OK

**Comment by dashixiongsk** [comment]
Sorry, the json format has been changed

**Comment by hanqingwu** [comment]
Can  you try example like this  ?
https://apisix.apache.org/docs/apisix/plugins/forward-auth/#example-usage
I have try to do like it , it works fine .

**Comment by dashixiongsk** [comment]
@hanqingwu Yes, I have no problem with that example on my end. Now downgrade the version from 3.8.0 to 3.6.0, there's no problem

**Comment by UzairM** [comment]
thanks @dashixiongsk . I was having the same issue and successfully resolved it by downgrading to 3.6.0

**Comment by kayx23** [comment]
So is there a bug with 3.8.0? If not, can this issue be closed?

**Comment by kayx23** [comment]
@dashixiongsk to clarify - this timeout error is encountered when setting `"request_method": "POST"`?

**Comment by luodadaRyan** [comment]
我这边遇到了同样的问题，验证3.6.0 及 3.7.0 是可以的，但是3.8.0以上就会出现同样的问题，POST转发到下游，下游无法读取

**Comment by GongchuangSu** [comment]
The temporary solution can be implemented by modifying and overriding the source forward-auth.lua script
```lua
    local httpc = http.new()
    httpc:set_timeout(conf.timeout)
--     if params.method == "POST" then
--         local client_body_reader, err = httpc:get_client_body_reader()
--         if client_body_reader then
--             params.body = client_body_reader
--         else
--             core.log.warn("failed to get client_body_reader. err: ", err,
--             " using core.request.get_body() instead")
--             params.body = core.request.get_body()
--         end
--     end
    if params.method == "POST" then
        params.body = core.request.get_body()
    end
```

**Comment by LLLLimbo** [comment]
Has this issue not been fixed yet ?

**Comment by Baoyuantop** [comment]
Duplicate of https://github.com/apache/apisix/issues/11200
